src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -330,14 +330,25 @@ def _rbd_image(cls, ioctx, pool_name, namespace, image_name):
             return stat
 
     @classmethod
-    def _rbd_image_names(cls, ioctx):
+    def _rbd_image_refs(cls, ioctx):
         rbd_inst = rbd.RBD()
-        return rbd_inst.list(ioctx)
+        return rbd_inst.list2(ioctx)
 
     @classmethod
     def _rbd_image_stat(cls, ioctx, pool_name, namespace, image_name):
         return cls._rbd_image(ioctx, pool_name, namespace, image_name)
 
+    @classmethod
+    def _rbd_image_stat_removing(cls, ioctx, pool_name, namespace, image_id):
+        rbd_inst = rbd.RBD()
+        img = rbd_inst.trash_get(ioctx, image_id)
+        img['unique_id'] = get_image_spec(pool_name, namespace, image_id)
+        img['pool_name'] = pool_name
+        img['namespace'] = namespace
+        img['deletion_time'] = "{}Z".format(img['deletion_time'].isoformat())
+        img['deferment_end_time'] = "{}Z".format(img['deferment_end_time'].isoformat())
+        return img
```

Comment:
I think this function is misnamed.  It should either make sure that the image is indeed in `REMOVING` state and fail otherwise or be renamed to not mention `removing`.

Reply:
It's the same situation as above: "So I guess I should filter for "USER" RBDs, so they won't be returned?" isn't it?

Reply:
Yes, if you are going to keep `removing` in the name of the function, I think it should filter for `REMOVING` and error out on any other state.

Or, rename this function to something like `_rbd_image_stat_trash()` so the name doesn't suggest any specificity and append to `result` after testing for `source == REMOVING`.